examples/router/webpack.config.js

```diff
@@ -34,7 +35,8 @@ module.exports = {
     ]
   },
   plugins: [],
-  externals: Object.assign({}, externals, {
+  externals: [...externals.filter(e => typeof e === 'function'), {
```

Comment:
you are right, the previous code wouldn't have worked well, because `require('frint-config').externals` is an Array now. nice catch, @ascariandrea!

Could you please try this one out, and see if it works:

```js
const externals = require('frint-config').externals;

module.exports = {
  externals: externals // covers RxJS, Lodash, React and PropTypes
    .concat([ // now FrintJS packages
      {
        'frint': {
          'root': 'Frint',
          'commonjs': 'frint',
          'commonjs2': 'frint',
          'amd': 'frint'
        },
        'frint-store': {
          'root': 'FrintStore',
          'commonjs': 'frint-store',
          'commonjs2': 'frint-store',
          'amd': 'frint-store'
        },
        'frint-model': {
          'root': 'FrintModel',
          'commonjs': 'frint-model',
          'commonjs2': 'frint-model',
          'amd': 'frint-model'
        },
        'frint-data': {
          'root': 'FrintData',
          'commonjs': 'frint-data',
          'commonjs2': 'frint-data',
          'amd': 'frint-data'
        },
        'frint-react': {
          'root': 'FrintReact',
          'commonjs': 'frint-react',
          'commonjs2': 'frint-react',
          'amd': 'frint-react'
        },
        'frint-router': {
          'root': 'FrintRouter',
          'commonjs': 'frint-router',
          'commonjs2': 'frint-router',
          'amd': 'frint-router'
        },
        'frint-router-react': {
          'root': 'FrintRouterReact',
          'commonjs': 'frint-router-react',
          'commonjs2': 'frint-router-react',
          'amd': 'frint-router-react'
        },
      },
    ]);
};
```

Reply:
Yep, the proposed way should work as well, however, the duplicated keys won't be overridden.
Do we need to override keys in some cases?

Reply:
`frint-config` was primarily built for handling the dists generation of all our packages.

if overriding is needed or becomes an issue, I wouldn't mind if examples drop `frint-config` usage completely.

then the webpack config could look like this:

```js
const webpackRxjsExternals = require('webpack-rxjs-externals');

module.exports = {
  externals: [
    // rxjs/*
    webpackRxjsExternals(),

    // lodash/*
    function (context, request, callback) {
      if (request.startsWith('lodash/')) {
        const subModule = request.split('/')[1];

        return callback(null, {
          root: ['_', subModule],
          commonjs: request,
          commonjs2: request,
          amd: request,
        });
      }

      return callback();
    },

      // full imports
    {
      'lodash': {
        root: '_',
        commonjs: 'lodash',
        commonjs2: 'lodash',
        amd: 'lodash',
      },
      'rxjs': {
        root: 'Rx',
        commonjs: 'rxjs',
        commonjs2: 'rxjs',
        amd: 'rxjs',
      },
      'react': {
        root: 'React',
        commonjs: 'react',
        commonjs2: 'react',
        amd: 'react',
      },
      'react-dom': {
        root: 'ReactDOM',
        commonjs: 'react-dom',
        commonjs2: 'react-dom',
        amd: 'react-dom',
      },
      'prop-types': {
        root: 'PropTypes',
        commonjs: 'prop-types',
        commonjs2: 'prop-types',
        amd: 'prop-types',
      },
    },

    // frint imports
    {
      'frint': {
        'root': 'Frint',
        'commonjs': 'frint',
        'commonjs2': 'frint',
        'amd': 'frint'
      },
      'frint-store': {
        'root': 'FrintStore',
        'commonjs': 'frint-store',
        'commonjs2': 'frint-store',
        'amd': 'frint-store'
      },
      'frint-model': {
        'root': 'FrintModel',
        'commonjs': 'frint-model',
        'commonjs2': 'frint-model',
        'amd': 'frint-model'
      },
      'frint-data': {
        'root': 'FrintData',
        'commonjs': 'frint-data',
        'commonjs2': 'frint-data',
        'amd': 'frint-data'
      },
      'frint-react': {
        'root': 'FrintReact',
        'commonjs': 'frint-react',
        'commonjs2': 'frint-react',
        'amd': 'frint-react'
      },
      'frint-router': {
        'root': 'FrintRouter',
        'commonjs': 'frint-router',
        'commonjs2': 'frint-router',
        'amd': 'frint-router'
      },
      'frint-router-react': {
        'root': 'FrintRouterReact',
        'commonjs': 'frint-router-react',
        'commonjs2': 'frint-router-react',
        'amd': 'frint-router-react'
      },
    }
  ],
};
```

The config does end up quite big though, but at least it won't have any surprises if the developer wants to customize further.

Reply:
if we define ALL the externals in `frint-config`, including the `frint-*` packages, we risk not generating proper dists for the packages. our CDN users would face difficulties then.

for example, the dist of `frint-react` includes `frint-component-handlers` and `frint-component-utils` in it.

Reply:
What about an util file responsible to merge the `externals` from `frint-config` with others used in `examples`?
Then replace each `frint-config` import in `webpack.config.js`s with the import of the util file?
I think it's more maintainable on the long run

Reply:
if we can avoid functions, and just have arrays/objects coming from `frint-config`, that would be cool.

for example:

```js
// has RxJS, Lodash, React, PropType
const thirdPartyExternals = require('frint-config').thirdPartyExternals;

// all frint-* packages
const frintExternals = require('frint-config').frintExternals;

module.exports = {
  externals: [
    ...thirdPartyExternals,
    ...frintExternals,
  ],
};
```

Reply:
This was the first solution I came up with, but I think it would affect other parts of the project.
I would handle this "refactor" in a separate PR after having pointed out what we need to achieve.

Reply:
glad we are on the same page here!

to make sure things work without breaking anything, we can just do the following:

1. keep `require('frint-config').externals` as it is, since other packages depend on it for dists generation
1. add a new `require('frint-config').frintExternals` that lists all frint-* packages like the webpack config above in this comments thread (with root/amd/commonjs/etc keys)
1. and use both `externals` and `frintExternals` in examples' webpack config

please let me know how that sounds.

Reply:
Agree!

I would keep my last implementation with `getExternals` to define in one place how externals are merged.

Could be it good for you?

Reply:
I am afraid having a shared util in this monorepo for all examples wouldn't work well, since we have `frint-cli`.

Developers scaffold certain examples in their own machine like this:

```
$ frint init --example counter
```

Since the scaffolded example app in their machine would only contain a certain example's code, it will not be able to resolve `examples/getExternals.js` as it is in this PR's diff.